foundation/foundation-mda/src/main/resources/templates/data-delivery-pyspark/persist.py.vm

```diff
@@ -19,20 +19,33 @@
           SQLite: 'sqlite:///db.sqlite'
       :return: A connection object from the RDBMS connection pool.
       """
-      connection = self.rdbms_connection_pool.get_connection(db_url)
-      return connection
-    #end
+      return self._get_spark_connection(db_url)
+
+    def _get_spark_connection(self, name: str):
+        if self.session is not None:
+            return self.session
+        else:
+            self.session = self.create_spark_session(name)
+            return self.session
+
+   #end
 
    #if ($step.persist.shortCollectionType)
     def save_dataset(self, dataset: ${step.persist.shortCollectionType}[${step.persist.shortRecordType}], ${saveParam}) -> None:
    #elseif ($step.isPersistTypeRdbms())
     #set ($db_url = "db_url: str")
     def save_dataset(self, ${db_url}, dataset: ${step.persist.shortRecordType}, ${saveParam}) -> None:
       ${step.capitalizedName}Base.logger.info('Saving %s To RDBMS...' % self.descriptive_label)
-      rdbms_session = self.rdbms_connection_pool.get_connection(db_url)
-      engine = rdbms_session.get_bind()
-      dataset.createOrReplaceTempView(table_name)
-      rdbms_session.close()
+      rdbms_session = self.get_connection(db_url)
```

Comment:
S: There shouldn't be any need to get/create a new Spark session and then stop it - I think this also may result in some adverse side effects?  Using `dataset.write.jdbc(..)` is sufficient for writing a `DataFrame` to the configured JDBC datastore.